common/djangoapps/auth_exchange/views.py

```diff
@@ -44,14 +55,72 @@ def post(self, request, _backend):  # pylint: disable=arguments-differ
         scope = form.cleaned_data["scope"]
         client = form.cleaned_data["client"]
 
+        return self.exchange_access_token(request, user, scope, client)
+
+    def exchange_access_token(self, request, user, scope, client):
+        raise NotImplementedError()
+
+
+class DOPAccessTokenExchangeView(AccessTokenExchangeBase, AccessTokenView):
+
+    oauth2_adapter = adapters.DOPAdapter()
+
+    def exchange_access_token(self, request, user, scope, client):
         if constants.SINGLE_ACCESS_TOKEN:
             edx_access_token = self.get_access_token(request, user, scope, client)
         else:
             edx_access_token = self.create_access_token(request, user, scope, client)
-
         return self.access_token_response(edx_access_token)
 
 
+class DOTAccessTokenExchangeView(AccessTokenExchangeBase, TokenView):
+
+    oauth2_adapter = adapters.DOTAdapter()
+
+    def get(self, request, _backend):
+        return Response(status=400, data={
+            'error': 'invalid_request',
+            'error_description': 'Only POST requests allowed.',
+        })
+
+    def populate_request(self, request, user, scope, client):
+        request.user = user
+        if scope is not None:
+            request.scopes = [SCOPE_VALUE_DICT[scope]]
+        else:
+            request.scopes = []
+        request.client = client
+        request.state = None
+        request.refresh_token = None
+        request.extra_credentials = None
+        request.grant_type = 'password'
+
+    def error_response(self, form_errors):
+        """
+        Return an error response consisting of the errors in the form
+        """
+        return Response(status=400, data=form_errors)
+
+    def exchange_access_token(self, request, user, scope, client):
```

Comment:
They don't, because DOT stores tokens in different tables, but I think it does make sense to push the differentiated behavior further down.  Either into those two methods, or further, into the adapters.